code/go/0chain.net/smartcontract/dbs/event/process.go

```diff
@@ -198,72 +198,80 @@ func (edb *EventDb) addEventsWorker(ctx context.Context) {
 	for {
 		es := <-edb.eventsChannel
 
-		if es.round%edb.settings.PartitionChangePeriod == 0 {
-			edb.managePartitions(es.round)
-		}
+		func() {
+			defer func() {
+				if r := recover(); r != nil {
```

Comment:
Perhaps we should let the sharder panic if any error happens during the event process, otherwise the data in event db could be different from MPT. But since we are ignoring errors atm, we can do strict (panic) later.

Reply:
we do not panic now on any error in the events, it's a little bit committing to panic on every error, block is already finalized an we can nothing to do to recover events execution. for testing purpose though we can panic() just to track events errors more tensely